src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,144 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> head;
+    private Node<T> tail;
+
+    public MyLinkedList() {
+    }
+
     @Override
     public void add(T value) {
-
+        Node<T> newNode = tail;
+        tail = new Node<>(value, null, newNode);
+        if (newNode == null) {
+            head = tail;
+        } else {
+            newNode.next = tail;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+        } else {
+            checkPosition(index);
```

Comment:
Did you mean move checkPosition at the start? If so, I can't do this because, checkPosition checks in range (index < 0 || index >= size).  
In my implementation, if index == size, I'll add value in the end. 
In another case, I check the index for correctness and put the value on the position. Also, I cannot change the checkPosition because I use this check in other methods, for example, get().

If you can help me with this, I'll be happy :)